docs/aws/audit/elbmonitoring/rules/elb_check_unused.mdx

```diff
@@ -24,6 +24,288 @@ HITRUST, SOC2, NISTCSF
+
+1. **List All ELBs:**
+   First, you need to list all the existing ELBs in your account to identify which ones are currently in use and which ones are not.
+
+   ```sh
+   aws elb describe-load-balancers --query 'LoadBalancerDescriptions[*].LoadBalancerName'
+   ```
+
+2. **Check ELB Usage:**
+   For each ELB, check if there are any instances registered with it. An ELB with no registered instances is likely unused.
+
+   ```sh
+   aws elb describe-load-balancers --load-balancer-names <load-balancer-name> --query 'LoadBalancerDescriptions[*].Instances'
+   ```
+
+3. **Tag ELBs:**
+   Tag your ELBs with metadata that indicates their purpose and owner. This helps in identifying and managing ELBs more effectively.
+
+   ```sh
+   aws elb add-tags --load-balancer-names <load-balancer-name> --tags Key=Purpose,Value=<purpose> Key=Owner,Value=<owner>
+   ```
+
+4. **Automate Monitoring:**
+   Set up a scheduled AWS Lambda function or a CloudWatch rule to periodically check for unused ELBs and notify the relevant stakeholders. This can be done using a Python script executed by AWS Lambda.
+
+   Example Python script snippet for Lambda:
+   ```python
+   import boto3
+
+   def lambda_handler(event, context):
+       client = boto3.client('elb')
+       elbs = client.describe_load_balancers()['LoadBalancerDescriptions']
+       unused_elbs = [elb['LoadBalancerName'] for elb in elbs if not elb['Instances']]
+       
+       if unused_elbs:
+           # Notify stakeholders or take action
+           print(f"Unused ELBs: {unused_elbs}")
+   ```
+
+By following these steps, you can ensure that you are actively monitoring and managing your ELBs to prevent having unused ones in your AWS environment.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent unused Elastic Load Balancers (ELBs) in AWS using Python scripts, you can follow these steps:
+
+1. **Set Up AWS SDK (Boto3) and Authentication:**
+   - Install the Boto3 library if you haven't already.
+   - Configure your AWS credentials.
+
+   ```bash
+   pip install boto3
+   ```
+
+   ```python
+   import boto3
+
+   # Initialize a session using Amazon EC2
+   session = boto3.Session(
+       aws_access_key_id='YOUR_ACCESS_KEY',
+       aws_secret_access_key='YOUR_SECRET_KEY',
+       region_name='YOUR_REGION'
+   )
+
+   # Initialize the ELB client
+   elb_client = session.client('elb')
+   ```
+
+2. **List All ELBs and Check for Unused ELBs:**
+   - Retrieve all ELBs and check if they have any instances attached.
+
+   ```python
+   def get_unused_elbs(elb_client):
+       unused_elbs = []
+       elbs = elb_client.describe_load_balancers()
+       for elb in elbs['LoadBalancerDescriptions']:
+           if not elb['Instances']:
+               unused_elbs.append(elb['LoadBalancerName'])
+       return unused_elbs
+
+   unused_elbs = get_unused_elbs(elb_client)
+   print("Unused ELBs:", unused_elbs)
+   ```
+
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>